third_party/zuho/zuho.js

```diff
@@ -248,8 +249,32 @@ export class Renderer {
     const gl = this.gl;
     gl.linkProgram(prog);
     const log = gl.getProgramInfoLog(prog);
-    if(log.length > 0) {
+    if (log.length > 0) {
       console.log(log);
     }
   }
+
+  /** @private */
+  euler_(heading, pitch, roll) {
+    const te = Matrix.identity(3);
+    const x = -roll, y = -pitch, z = heading;
+
+    const a = Math.cos(x), b = Math.sin(x);
+    const c = Math.cos(y), d = Math.sin(y);
+    const e = Math.cos(z), f = Math.sin(z);
+
+    const ae = a * e, af = a * f, be = b * e, bf = b * f;
+
+    te[0] = c * e;
+    te[3] = -c * f;
+    te[6] = d;
+    te[1] = af + be * d;
+    te[4] = ae - bf * d;
+    te[7] = -b * c;
+    te[2] = bf - ae * d;
+    te[5] = be + af * d;
+    te[8] = a * c;
+
+    return te;
+  }
 }
```

Comment:
nit: newline at end of file (unless this is intentional)

Reply:
Done